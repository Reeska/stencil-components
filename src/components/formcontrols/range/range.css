:host {
  align-items: center;
  display: flex;
  flex: 1 1 auto;
}

input {
  flex: 1 1 0%;
}

input[type=range] {
  -webkit-appearance: none;
  border: none;
  margin: 0;
  padding-left: 0;
  padding-right: 0;
}

input[type=range]::-webkit-slider-runnable-track {
  background: var(--evb-primary-color);
  border: none;
  border-radius: .25rem;
  cursor: pointer;
  height: .25rem;
  transition-duration: .2s;
  width: 100%;
}

input[type=range]::-webkit-slider-thumb {
  background: var(--evb-primary-color);
  border: 1px solid #fff;
  border-radius: 50%;
  cursor: pointer;
  height: 1rem;
  -webkit-appearance: none;
  margin-top: -0.5em;
  width: 1rem;
}

input[type=range]::-moz-range-track {
  background: var(--evb-primary-color);
  border: none;
  cursor: pointer;
  height: .25rem;
  transition-duration: .2s;
  border-radius: 25px;
  width: 100%;
}

input[type=range]::-moz-range-thumb {
  background: var(--evb-primary-color);
  border: 1px solid #fff;
  border-radius: 50%;
  cursor: pointer;
  height: 1rem;
  width: 1rem;
}

input[type=range]::-ms-track {
  background: transparent;
  border-color: transparent;
  border-width: 1px;
  color: transparent;
  cursor: pointer;
  height: .25rem;
  padding: .5rem;
  transition-duration: .2s;
  width: 100%;
}

input[type=range]::-ms-fill-lower {
  background: var(--evb-primary-color);
  border: 0 solid #000101;
  border-radius: 50;
}
input[type=range]::-ms-fill-upper {
  background: lighten(var(--evb-primary-color), 25%);
  border: 0 solid #000101;
  border-radius: 50;
}

input[type=range]::-ms-thumb {
  background: var(--evb-primary-color);
  border: 1px solid #fff;
  border-radius: 50%;
  cursor: pointer;
  height: 1rem;
  margin-top: -.1375em;
  width: 1rem;
}

input[type=range]:focus {
  outline: none;
}

input[type=range]:focus::-webkit-slider-runnable-track {
  background: var(--evb-primary-color-darker);
}

input[type=range]:focus::-ms-fill-lower {
  background: var(--evb-primary-color);
}

input[type=range]:focus::-ms-fill-upper {
  background: var(--evb-primary-color);
}
