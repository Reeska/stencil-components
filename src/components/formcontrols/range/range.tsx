import { Component, ComponentInterface, Element, Event, EventEmitter, h, Prop } from '@stencil/core';

@Component({
  tag: 'evb-range',
  styleUrl: 'range.css',
  shadow: true
})
export class EvbRange implements ComponentInterface {

  @Prop({ mutable: true }) value: number;
  @Prop() min: number;
  @Prop() max: number;
  @Prop() step: number;
  @Prop() disabled = false;

  @Event() evbBlur: EventEmitter<void>;
  @Event() evbChange: EventEmitter<number>;
  @Event() evbFocus: EventEmitter<void>;
  @Event() evbInput: EventEmitter<number>;

  @Element() host: HTMLElement;

  get inputValue(): number {
    const input: HTMLInputElement = this.host.shadowRoot.querySelector('input[type=range]');
    return input.valueAsNumber;
  }

  render() {
    return (
      <input type='range'
        value={this.value}
        disabled={this.disabled}
        min={this.min}
        max={this.max}
        step={this.step}
        onBlur={() => this.evbBlur.emit()}
        onChange={() => this.evbChange.emit(this.inputValue)}
        onFocus={() => this.evbFocus.emit()}
        onInput={() => this.evbInput.emit(this.inputValue)} />
    );
  }
}
