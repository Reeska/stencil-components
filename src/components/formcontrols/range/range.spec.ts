import { EvbRange } from './range';

describe('EvbRange', () => {
  let component: EvbRange;

  beforeEach(() => component = new EvbRange());

  it('should render', () => {
    expect(component.render()).toMatchSnapshot();
  });
});
