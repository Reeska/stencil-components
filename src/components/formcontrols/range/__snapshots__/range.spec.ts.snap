// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvbRange should render 1`] = `
Object {
  "$attrs$": Object {
    "max": undefined,
    "min": undefined,
    "onBlur": [Function],
    "onChange": [Function],
    "onFocus": [Function],
    "onInput": [Function],
    "step": undefined,
    "type": "range",
    "value": undefined,
  },
  "$children$": null,
  "$elm$": null,
  "$flags$": 0,
  "$key$": null,
  "$name$": null,
  "$tag$": "input",
  "$text$": null,
}
`;
